Jasmine/Tetris/ViewModels/TetrisGameViewModelProtocol.swift

```diff
@@ -29,4 +29,9 @@ protocol TetrisViewModelProtocol: BaseViewModelProtocol {
     ///
     /// - Parameter coordinate: the position to land the tile.
     func landFallingTile(at coordinate: Coordinate)
+
+    /// Swaps the current tile with the upcoming tile at `index`
+    ///
+    /// - Parameter index: the index of `upcomingTiles` to be swapped
+    func swapCurrentTileWithUpcomingTile(at index: Int)
```

Comment:
Prefer:
swapCurrentTile(withUpcomingAt index: Int)
Or something similar so it looks like
swapCurrentTile(withUpcomingAt: 1)

And the implementer can use "index" instead of "at"